docs/8_prd_workflow_architecture.md

```diff
@@ -0,0 +1,618 @@
+
+### Phase 5: Finalization
+- **Complete** workflow and mark as finalized
+
+## Workflow Diagram
+
+```mermaid
+graph TD
+    Start[START] --> Init[Magi Initialization]
+    Init --> FB[Feature Brief Generation]
+    FB --> IR[Initial Requirements Generation]
+    IR --> RR[Requirements Review]
+    RR --> GA[Gap Analysis]
+    
+    GA -->|Score < threshold| IC[Iteration Control]
+    GA -->|Score >= threshold| PG[PRD Generation]
+    
+    IC -->|Continue| GR[Gap Requirements Generation]
+    IC -->|Stop| PG
+    
+    GR --> RR
+    
+    PG --> PR[PRD Review]
+    
+    PR -->|Approved| F[Finalization]
+    PR -->|Not Approved| PG
+    
+    F --> End[END]
+```
+
+## Architecture Components
+
+### Node Types
+
+There are two base node classes in the PRD workflow:
+
+> NOTE: there is a lot of overlap from the normal magen workflow here, these will be refactored in the future
+> to share the same abstracted orchestrator/nodes.
+
+1. **`PRDBaseNode`** - Base class for all nodes
+   - Provides standard node name and execution interface
+   - Used for state transformation without tool invocation
+
+2. **`PRDAbstractToolNode`** - Extends `PRDBaseNode`
+   - Adds tool execution capabilities
+   - Provides standardized logging and error handling
+   - Includes `executeToolWithLogging()` for safe tool invocation
+
+## Node Details
+
+### 1. Magi Initialization Node
+**Class:** `PRDMagiInitializationNode`  
+**Type:** Base Node (no tools)
+
+**Purpose:** Initializes the PRD workflow by validating inputs and creating the necessary directory structure.
+
+**Responsibilities:**
+- Validates that `projectPath` is provided in user input
+- Validates that `originalUserUtterance` is provided
+- Ensures the `magi-sdd` directory exists in the project
+- Returns validated paths and user utterance
+
+**Key State Updates:**
+- Sets `projectPath` - the root project directory
+- Sets `originalUserUtterance` - the user's original request
+- Sets `magiSddPath` - path to the magi-sdd directory
+
+**Error Handling:**
+- Throws error if `projectPath` is missing
+- Throws error if `originalUserUtterance` is missing
+- Throws error if directory creation fails
+
+---
+
+### 2. Feature Brief Generation Node
+**Class:** `PRDFeatureBriefGenerationNode`  
+**Type:** Tool Node  
+**Tool:** `magi-prd-feature-brief`
+
+**Purpose:** Generates a structured feature brief from the user's original utterance.
+
+**Responsibilities:**
+- Calls the feature brief generation tool
+- Retrieves existing feature IDs to ensure uniqueness
+- Creates feature directory structure
+- Writes feature brief to disk
+- Generates recommended feature ID
+
+**Tool Input:**
+```typescript
+{
+  userUtterance: string,
+  currentFeatureIds: string[]
+}
+```
+
+**Tool Output:**
+```typescript
+{
+  featureBriefMarkdown: string,
+  recommendedFeatureId: string
+}
+```
+
+**Key State Updates:**
+- Sets `featureId` - unique identifier for the feature
+- Sets `featureDirectoryPath` - path to feature directory
```

Comment:
How does this relate to the `{magiSddPath}/{featureId}` path below? Wondering if we should store it as state, vs. infer it from other state.